omegatrade-app/frontend/src/app/components/stock-dashboard/stock-dashboard.component.ts

```diff
@@ -0,0 +1,218 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Chart, StockChart } from 'angular-highcharts';
+import { ActivatedRoute } from '@angular/router';
+import { RestService } from '../../services/rest.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { interval, Subscription } from "rxjs";
+import { take } from 'rxjs/operators';
+import { SnackBarService } from '../../services/snackbar.service';
+
+@Component({
+  selector: 'app-stock-dashboard',
+  templateUrl: './stock-dashboard.component.html',
+  styleUrls: ['./stock-dashboard.component.css']
+})
+export class StockDashboardComponent implements OnInit , OnDestroy{
+
+  selectedCompany: string = "";
+  lastUpdatedTime;
+  stock: StockChart;
+  companies: any;
+  loader: boolean = false;
+  subscription: Subscription;
+  timerIds = [];
+  
+  constructor(private snackBarService: SnackBarService, private router: ActivatedRoute, private _snackBar: MatSnackBar, private restService: RestService) {
+  }
+
+  /**
+   *  Function to Initiate component.
+   *  Initiating company lists
+   */
+  ngOnInit(): void {
+    this.selectedCompany = this.router.snapshot.queryParamMap.get('companyId');
+    this.getCompanies();
+  }
+
+  /**
+   *  Function to get all companies.
+   *  @returns  {null}
+   */
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.companies = response.data;
+            if (this.companies && this.companies.length > 0) {
+              if(!this.selectedCompany || this.selectedCompany === "")
+                this.selectedCompany = this.companies[0].companyId;
+              this.getStockData();
+            }
+          }
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.clearAllTimeOuts();
+          this.loader = false;
+        });
+  }
+
+  /**
+   *  Function to get all stockDatas for selected company.
+   * 
+   *  @returns  {null}
+   */
+  getStockData() {
+    if (this.selectedCompany) {
+      this.loader = true;
+      this.restService.getData(`companies/dashboard/${this.selectedCompany}`)
+        .pipe(take(1))
+        .subscribe(
+          response => {
+            if (response && response.success) {
```

Comment:
What happens if `response && response.success` is falsy?

Reply:
I've added this condition `response && response. success` so that it will not affect the further logic in case of any failure.
The observable invokes the first callback response when the HTTP request successfully returns a response.
In the backend, I've added `{success: true}`  for successful events. Any unsuccessful events will be handled in error callback.